index.js

```diff
@@ -96,6 +96,7 @@ let executeHooksWithArgs = (hooks = [], args) => {
                 commandIsRunning = _commandIsRunning
             }).run()
         } catch (e) {
+            reject(e)
```

Comment:
consider the following test suite:

```js
describe('suite', () => {
    before(() => {
        throw new Error('stop test right now!')
    })
    
    it('is a test which should not run', () => {})
})
```

Vanilla Mocha gets the error from `before` and stops the suite. With wdio-sync wrapper around the error will be swallowed and Mocha will run test case like `before` hook went fine.

Do you really think that this is okay to change default behavior of Mocha tests?